x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/huggingface/HuggingFaceBaseService.java

```diff
@@ -127,7 +134,25 @@ protected void doChunkedInfer(
         ChunkingOptions chunkingOptions,
         ActionListener<List<ChunkedInferenceServiceResults>> listener
     ) {
-        listener.onFailure(new UnsupportedOperationException("Chunked inference not implemented for Hugging Face"));
+        ActionListener<InferenceServiceResults> inferListener = listener.delegateFailureAndWrap(
+            (delegate, response) -> delegate.onResponse(translateToChunkedResults(input, response))
+        );
+
+        doInfer(model, input, taskSettings, inputType, inferListener);
     }
 
+    private static List<ChunkedInferenceServiceResults> translateToChunkedResults(
+        List<String> inputs,
+        InferenceServiceResults inferenceResults
+    ) {
+        if (inferenceResults instanceof TextEmbeddingResults textEmbeddingResults) {
```

Comment:
Technically we could make this more strict because a task type of `sparse_embedding` should never return `TextEmbeddingResults`, only `SparseEmbeddingResults`. I figured this code is temporary so I didn't go down that route.